docs/4.x/archiving.md

```diff
@@ -137,10 +153,165 @@ For example, the *UserSettings* plugin uses one record to hold browser details o
+
+## Archive Purging
+
+Archives that are invalidated or no longer require processing do not need to be stored, so they are deleted. Old and
+errored archives are deleted in two places:
+
+* right after a new archive is finalized, older archives for the same site, period and segment are deleted
+* regularly through a scheduled task in the `CoreAdminHome` plugin, older and errored archives are deleted
+
+The specifics of purging are straightforward. `idarchive`s of archives that are safe to delete are queried, then
+rows with that `idarchive` in the respective numeric and blob tables are deleted.
+
+### Other instances where archives are deleted
+
+Archives can also be deleted in other contexts. For example, when a segment is deleted, we no longer need the
+archives for that segment, so we delete them.
+
+Plugins can delete archives themselves by using the `ArchivePurger` class.
+
+## Partial Archives
+
+Partial archives are a special type of archive that only contain one or a few reports of one plugin, rather than every report
+for one plugin (or every report for all plugins). These archives have done flag value of `ArchiveWriter::DONE_PARTIAL`.
+
+Currently, they are only created when plugins want to archive a single report in the past.
+
+### For plugins: supporting partial archives
+
+Partial archives are not automatically created when a single report is requested for archiving. Plugin `Archiver`s have to
+specifically support this workflow like so:
+
+```
+class Archiver extends \Piwik\Plugin\Archiver
+{
+    // ...
+
+    public function __construct(ArchiveProcessor $processor)
+    {
+        parent::__construct($processor);
+
+        // if a single report is requested, mark the archive we're creating as partial
+        $this->requestedReport = $processor->getParams()->getArchiveOnlyReport();
+        if ($this->requestedReport) {
+            $processor->getParams()->setIsPartialArchive(true);
+        }
+    }
+
+    public function aggregateDayReport()
+    {
+        if ($this->isArchiving('MyPlugin_mySpecificReport')) {
+            $maxRowsInTable = Config::getInstance()->General['datatable_archiving_maximum_rows_standard'];j
+
+            $dataTable = // ... build by aggregating visits ...
+            $serializedData = $dataTable->getSerialized(
+                $maxRowsInTable,
+                $maxRowsInSubtable = $maxRowsInTable,
+                $columnToSortBy = Metrics::INDEX_NB_VISITS
+            );
+
+            $archiveProcessor->insertBlobRecords('MyPlugin_mySpecificReport', $serializedData);
+        }
+    }
+
+    public function aggregateMultipleReports()
+    {
+        if ($this->isArchiving('MyPlugin_mySpecificReport')) {
+            $this->getProcessor()->aggregateDataTableRecords(['MyPlugin_mySpecificReport']);
+        }
+    }
+
+    private function isArchiving(string $reportName)
```

Comment:
should we put this method maybe in the parent class so not every one has to add it?

Reply:
Yeah, though if it's not used in conjunction w/ the other code, it'll just introduce bugs. I thought it was better to require users to be explicit about how they implement partial archiving support, and not just use it because they see it.

Reply:
OK no big preference. Just would avoid needing it in several archivers (since we'll add it to funnels, usersflow, custom reports, ...) but it's also not complex method so can be added into the archiver again